### ISSUE TITLE
Drop setuptools

### ISSUE THREAD

**Issue by lasers** [ultrabug/py3status#2196]
**Describe the bug**

```python
# Formatted for readable.
alice@machine:~/src/py3status$ sudo python3 setup.py develop

running develop
/usr/lib/python3/dist-packages/setuptools/command/easy_install.py:146:
    EasyInstallDeprecationWarning: easy_install command is deprecated.
    Use build and pip and other standards-based tools.
   warnings.warn(

/usr/lib/python3/dist-packages/setuptools/command/install.py:34:
    SetuptoolsDeprecationWarning: setup.py install is deprecated.
    Use build and pip and other standards-based tools.
   warnings.warn(

# Same thing with "sudo python3 setup.py install".
```

**Additional context**

DEPRECATED! DEPRECATED! DEPRECATED!

**Comment by lasers** [comment]
**Describe the bug**

```shell
# Formatted for readable.
❯ pip install py3status
Collecting py3status
  Downloading py3status-3.50.tar.gz (399 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 399.8/399.8 kB 6.2 MB/s eta 0:00:00
  Preparing metadata (setup.py) ... done
Requirement already satisfied: setuptools in ./.venv/lib/python3.11/site-packages (from py3status) (65.5.0)
Installing collected packages: py3status
  DEPRECATION: py3status is being installed using the legacy 'setup.py install' method, because it does not
have a 'pyproject.toml' and the 'wheel' package is not installed. pip 23.1 will enforce this behaviour change.
A possible replacement is to enable the '--use-pep517' option. Discussion can be found at
https://github.com/pypa/pip/issues/8559
  Running setup.py install for py3status ... done
Successfully installed py3status-3.50

[notice] A new release of pip available: 22.3.1 -> 23.1.2
[notice] To update, run: pip install --upgrade pip
```

**Additional context**

You are on NOTICE!

**Comment by ultrabug** [comment]
I switched to hatch now, thanks
